pkg/standalone-utils/contracts/ERC4626CowSwapFeeBurner.sol

```diff
@@ -52,32 +55,45 @@ contract ERC4626CowSwapFeeBurner is CowSwapFeeBurner {
         IERC20 feeToken,
         uint256 exactFeeTokenAmountIn,
         IERC20 targetToken,
-        uint256 minTargetTokenAmountOut,
+        uint256 encodedMinAmountsOut,
         address recipient,
         uint256 deadline
     ) external override onlyProtocolFeeSweeper nonReentrant {
+        IERC4626 erc4626Token = IERC4626(address(feeToken));
+        IERC20 underlyingToken = IERC20(erc4626Token.asset());
+
         // In this case we first pull the wrapped token, unwrap, and then proceed to burn by creating an order for
         // the underlying token.
-        feeToken.safeTransferFrom(msg.sender, address(this), exactFeeTokenAmountIn);
+        IERC20(address(erc4626Token)).safeTransferFrom(msg.sender, address(this), exactFeeTokenAmountIn);
 
-        // Redeem and overwrite inputs with new asset and unwrapped amount.
-        IERC4626 erc4626Token = IERC4626(address(feeToken));
-        feeToken = IERC20(erc4626Token.asset());
-        exactFeeTokenAmountIn = erc4626Token.redeem(exactFeeTokenAmountIn, address(this), address(this));
+        (uint256 minTargetTokenAmountOut, uint256 minERC4626AmountOut) = PackedTokenBalance.fromPackedBalance(
+            bytes32(encodedMinAmountsOut)
+        );
+
+        uint256 feeTokenBalanceBefore = underlyingToken.balanceOf(address(this));
+
+        erc4626Token.redeem(exactFeeTokenAmountIn, address(this), address(this));
+
+        uint256 feeTokenBalanceAfter = underlyingToken.balanceOf(address(this));
+        exactFeeTokenAmountIn = feeTokenBalanceAfter - feeTokenBalanceBefore;
+
+        if (exactFeeTokenAmountIn < minERC4626AmountOut) {
+            revert AmountOutBelowMin(underlyingToken, exactFeeTokenAmountIn, minERC4626AmountOut);
+        }
```

Comment:
Two comments here:
- ERC4626 should return the assets redeemed. It is true that measuring the balances should be accurate, but unless the ERC4626 is non-compliant it shouldn't be strictly necessary. I'm fine with leaving it either way.
- If `minERC4626AmountOut` is 0 and `exactFeeTokenAmountIn` is 0 this should revert as well. The code as is wouldn't prevent the issue we've seen yesterday unless the `minERC4626AmountOut` is encoded, which should be optional